pandas/core/arrays/_mixins.py

```diff
@@ -252,6 +258,9 @@ def shift(self, periods: int = 1, fill_value=None) -> Self:
         return self._from_backing_data(new_values)
 
     def __setitem__(self, key, value) -> None:
+        if self._readonly:
+            raise ValueError("Cannot modify readonly array")
```

Comment:
As mentioned in the dev call, here's an example where what if `self._ndarray` was made read only already so we didn't need to manually raise an exception here?

Not sure if it's feasible, for numpy-backed arrays to have a 


```python
def _as_readonly(self) -> None
    self._ndarray.flags.writable = False
```

method to convert an EA's state to read-only